pkg/store/fake_store_test.go

```diff
@@ -263,6 +264,193 @@ func TestFakeStoreIngressV1(t *testing.T) {
 	assert.Len(store.ListIngressesV1beta1(), 0)
 }
 
+func TestFakeStoreIngressClassParams(t *testing.T) {
+	assert := assert.New(t)
+	apiGroup := ingressClassParamsAPIGroup
+	unrelatedAPIGroup := "unrelated.com"
+
+	cases := []struct {
+		name               string
+		ingressClassParams []*kongv1alpha1.IngressClassParams
+		ingressClasses     []*networkingv1.IngressClass
+		shouldExist        bool
+	}{
+		{
+			name:               "When there are no ingress classes defined",
+			ingressClassParams: []*kongv1alpha1.IngressClassParams{},
+			ingressClasses:     []*networkingv1.IngressClass{},
```

Comment:
```suggestion
```

two comments here:
- `nil` slices are better than empty literal slices, because the latter occupy memory unnecessarily
- there's no need to initialize empty slices explicitly

https://medium.com/@habibridho/golang-nil-vs-empty-slice-87fd51c0a4d

Please follow this style throughout the PR.